Feature: Add nav component

georgehossa opened the pull request:
This adds the navbar component, I need help with the transition when is contracted, and I don't know if I do it well.

Fixes #5

georgehossa committed:
Feature: Add nav component

georgehossa committed:
Feature: Add base layout

wolftrax5 commented on src/components/NavBar/index.js:
```diff
@@ -0,0 +1,51 @@
+          {expanded ? <img src={logoFull} alt='Hope Tracker' /> : <i className='icon__brand'></i> }
+        </div>
+        <ul className='NavBar__list'>
+          <li className="NavBar__item">
```
This can be a reusable component, a have a state or props in NabBar type array like `[{to:"/global", label: "Global"}, {to:"/", label: "Home"}]` and iterate with a map function and return the NavItem component

andresV74 replied:
I agarre with @wolftrax5

andresV74 reviewed (commented):
I've just started to see how it works and I got this reviews by far:
1. I think it would be better if the menu is collapsed when the page is loaded.
2. I would change position sticky to fixed.
3. I got your point, I think the transition could be more fluid, but I still have to look more deeply at the code to know how to do it (I hope I can review it by this weekend).

georgehossa committed:
Merge branch 'feature/base-layout' into feature/nav-section

georgehossa commented:
> I've just started to see how it works and I got this reviews by far:
> 
> 1. I think it would be better if the menu is collapsed when the page is loaded.
> 2. I would change position sticky to fixed.
> 3. I got your point, I think the transition could be more fluid, but I still have to look more deeply at the code to know how to do it (I hope I can review it by this weekend).

georgehossa committed:
Add Eslint and prettier configuration

georgehossa committed:
Refactor Layout component, rename it

georgehossa committed:
Fix Wrapper component change in previous commit

georgehossa committed:
Refactor NavBar component

- Add links info via component props and extract link structure to render links in a map function.
- Change logo structure to take a better handle of state change.

georgehossa committed:
Merge branch 'develop' into feature/nav-section
